sensor/init-tls-certs/main.go

```diff
@@ -0,0 +1,167 @@
+}
+
+func main() {
+	flag.Parse()
+
+	log.Println("New certs directory:", newSourceDir)
+	log.Println("Legacy certs directory:", legacySourceDir)
+	log.Println("Destination directory:", destinationDir)
+
+	realDest, err := sanityCheckDestination()
+	if err != nil {
+		log.Fatalf("Cannot check destination directory %q: %s", destinationDir, err)
+	}
+	log.Printf("Destination directory %q looks sane.", destinationDir)
+
+	timeout := 5 * time.Minute
+	ctx, cancel := context.WithTimeout(context.Background(), timeout)
+	defer cancel()
+
+	files, err := waitForSource(ctx)
+	if err != nil {
+		log.Fatalf("Failed to load certificates: %v", err)
+	}
+
+	if err = copyFiles(files, realDest); err != nil {
+		log.Fatalf("Cannot copy files: %v", err)
+	}
+}
+
+func copyFiles(files []string, destDir string) error {
+	for _, file := range files {
+		content, err := os.ReadFile(file)
+		if err != nil {
+			return err
+		}
+		destPath := path.Join(destDir, path.Base(file))
+		perm := os.FileMode(0600) // 0600 is required by Postgres (used by scanner-db)
+		if err = os.WriteFile(destPath, content, perm); err != nil {
+			return err
+		}
+		log.Printf("Copied %q to %q", file, destPath)
+	}
+	return nil
+}
+
+func waitForSource(ctx context.Context) ([]string, error) {
+	log.Printf("Looking for files in the source directory %q.", legacySourceDir)
+	for {
+		select {
+		case <-ctx.Done():
+			return nil, fmt.Errorf("context expired while waiting for certificates: %w", ctx.Err())
+		default:
+			// Check new certificates first
+			files, err := findFiles(newSourceDir)
+			if err != nil {
+				log.Printf("Error checking certificates in %q: %s", newSourceDir, err)
+			} else {
+				log.Printf("Using %d new certificate files from %q.", len(files), newSourceDir)
+				return files, nil
+			}
+
+			// Fall back to legacy certificates
+			files, err = findFiles(legacySourceDir)
+			if err != nil {
+				log.Printf("Error checking legacy certificates in %q: %s", legacySourceDir, err)
+			} else {
+				log.Printf("Using %d legacy certificates from %q.", len(files), legacySourceDir)
+				return files, nil
+			}
+
+			log.Printf("No certificates found. Retrying...")
+			time.Sleep(5 * time.Second)
+		}
+	}
+}
+
+func findFiles(sourceDir string) ([]string, error) {
+	realSource, err := filepath.EvalSymlinks(sourceDir)
+	if err != nil {
+		return nil, err
+	}
+
+	log.Printf("Walking %q.", realSource)
+	var files []string
+	err = filepath.WalkDir(realSource, func(path string, d fs.DirEntry, walkErr error) error {
+		if walkErr != nil {
+			log.Printf("Error accessing path %q: %s", path, walkErr)
+			return nil
+		}
+
+		base := filepath.Base(path)
+		if strings.HasPrefix(base, ".") {
+			if d.IsDir() {
+				log.Printf("Ignoring hidden dir %q", path)
+				return filepath.SkipDir
+			}
+
+			log.Printf("Ignoring hidden file %q", path)
+			return nil
+		}
+
+		realFile, err := filepath.EvalSymlinks(path)
```

Comment:
I also think it should work without evaluating symlinks. @porridge is there some corner case where this might be needed?

Reply:
@mclasmeier IMO the difference is if the symlinks are broken. In that case, this will just ignore the file.

If we remove the `filepath.EvalSymlinks` call, it would return this as a valid file, and then `copyFiles` would fail. So possibly the current approach is better.